### ISSUE TITLE
[Gamepad issues] R2\L2 does nothing and Opentk problems

### ISSUE THREAD

**Issue by InfamousKnight** [TASVideos/BizHawk#2912]
The secondary shoulder buttons do nothing on DirectInput/Xinput and Opentk is completely broken with gamepad.

To be clear, DirectInput works for all other keys other than secondary shoulder buttons while opentk is completely nonfunctional with gamepads.

Commit b29516e tested on linux

**Comment by YoshiRulz** [comment]
How were you using DirectInput on Linux? Though the option is available, it falls back to OpenTK.

Anyway, I'm having no problems with my xbone controller. What device are you trying to use?

**Comment by InfamousKnight** [comment]
Hmm, I'm using a xbone controller too. For some reason, DirectInput doesn't fall back to opentk on my end. I'm using wine to test the windows version. And wine seems to work with directinput except for shoulder buttons

**Comment by YoshiRulz** [comment]
Don't use WINE.

**Comment by InfamousKnight** [comment]
I did test it without wine. DirectInput doesnt fallback to opentk with EmuHawkMono and they are both disfunctional on linux end.

**Comment by YoshiRulz** [comment]
Oh okay. Well you should see something like this in stdout (run as `./EmuHawkMono.sh --mono-no-redirect`):
```
Initialising OpenTK GamePad: 03000000-5e04-0000-ea02-000001030000
OpenTK Mapping: Xbox One Wireless Controller (Model 1708)
Connected new gamepad #0, port#0 (Xbox One Wireless Controller (Model 1708)) via OpenTK
```
If there's nothing then OpenTK is failing. Which other programs are able to use the gamepad? Does it work [in the browser](https://gamepad-tester.com/) for example?

**Comment by InfamousKnight** [comment]
This is what shows up in the console log:

Initialising OpenTK GamePad: 03000000-c624-0000-3a54-000001010000
OpenTK Mapping: XInput Controller
Connected new gamepad #0, port#0 (XInput Controller) via OpenTK


And no, the browser doesn't detect it

**Comment by YoshiRulz** [comment]
ee0848d7d changes the logging a bit, now it will read either
```
Connected new gamepad as #0: port 0 mapped 03000000-c624-0000-3a54-000001010000 XInput Controller
```
or
```
Connected new gamepad as #0: port 0 unmapped 03000000-c624-0000-3a54-000001010000 XInput Controller
```
I have a feeling it will be the latter, as I don't see that UUID in [the database](https://github.com/opentk/opentk/blob/3.x/src/OpenTK/Input/GamePadConfigurationDatabase.cs). If so then it needs to be added there first. Why your revision of xbone controller is missing is a mystery to me, maybe it's just very rare.

**Comment by InfamousKnight** [comment]
Its neither:

sing OpenTK 3 for host input (keyboard + gamepads)
Connected new gamepad as #0: port #0 unmapped 03000000-c624-0000-3a54-000001010000 Unmapped Controller

**Comment by InfamousKnight** [comment]
Tested and it works!

**Comment by YoshiRulz** [comment]
The changes from today fixed the problem?

**Comment by InfamousKnight** [comment]
Yes!

**Comment by InfamousKnight** [comment]
To verify, I used the same controller on last commit as of writing this and if worked like a charm! Even the r2/l2 problem was fixed. This is strange because presumably opentk was to blame here correct?

**Comment by YoshiRulz** [comment]
No idea. I was trying to fix #2907, having given up on this issue.
